tests/testthat/test-ggplot-coord.R

```diff
@@ -0,0 +1,52 @@
+context("Fixed coordinates")
+
+# Expect trace function
+expect_traces <- function(gg, n_traces, name) {
+  stopifnot(is.ggplot(gg))
+  stopifnot(is.numeric(n.traces))
+  save_outputs(gg, paste0("coord_fixed-", name))
+  L <- gg2list(gg)
+  all_traces <- L$data
+  no_data <- sapply(all_traces, function(tr) {
+    is.null(tr[["x"]]) && is.null(tr[["y"]])
+  })
+  has_data <- all_traces[!no_data]
+  expect_equal(length(has_data), n_traces)
+  list(traces = has_data, layout = L$layout)
+}
+
+# Data where x ranges from 0-10, y ranges from 0-30
+set.seed(202)
+dat <- data.frame(xval = runif(40,0,10), yval = runif(40,0,30))
```

Comment:
Yes. ```set.seed()``` ensures we get the same data each time. I second
Carson's point. I used the set.seed with the same seed as used in the R
Cookbook. But I can use one of the datasets that come with R if it's
preferred.

Gotta go to graduation now! Will be back shortly.

-- Baobao Zhang

In tests/testthat/test-ggplot-coord.R
<https://github.com/ropensci/plotly/pull/223#discussion_r30524645>:

> +  stopifnot(is.ggplot(gg))
> +  stopifnot(is.numeric(n.traces))
> +  save_outputs(gg, paste0("coord_fixed-", name))
> +  L <- gg2list(gg)
> +  all_traces <- L$data
> +  no_data <- sapply(all_traces, function(tr) {
> +    is.null(tr[["x"]]) && is.null(tr[["y"]])
> +  })
> +  has_data <- all_traces[!no_data]
> +  expect_equal(length(has_data), n_traces)
> +  list(traces = has_data, layout = L$layout)
> +}
> +
> +# Data where x ranges from 0-10, y ranges from 0-30
> +set.seed(202)
> +dat <- data.frame(xval = runif(40,0,10), yval = runif(40,0,30))

I'm pretty sure set.seed() ensures we have the same set of numbers, so
image diffs should be meaningful. That being said, I prefer setting the
seed over copy/pasting numbers (I think it makes the tests easier to read).

Even better, use a built-in dataset such as mtcars, cars, iris, diamonds,
etc.

—
Reply to this email directly or view it on GitHub
<https://github.com/ropensci/plotly/pull/223/files#r30524645>.